### ISSUE TITLE
Distinguish between mobile and stationary observations

### ISSUE THREAD

**Issue by jo-asplin-met-no** [EURODEO/e-soh#29]
It would be useful to explicitly distinguish explicitly between mobile and stationary observations. Mobile observations are typically 
generated from sensors on cars, ships and aircraft, while stationary observations typically come from sensors on weather stations.

One use case that would be possible with such a distinction if that the EDR API can ask the datastore for all stationary time series inside a certain polygon, i.e. without having to download any observations (which could easily become resource intensive, since a robust solution would probably need to download the entire set of observations to ensure that all time series are represented). Assuming that a majority of the observations in E-SOH are likely to be stationary (?), this would be applying the old principle of _optimizing for the common case_.

From the perspective of the datastore, the position of an observation is currently represented only in the message ObsMetadata in datastore.proto (as either geo_point or geo_polygon (the latter currently not supported)). This means that mobile observations are supported by definition, since the observation can potentially change between observations within the same time series (remember that the datastore defines a time series as a unique combination of metadata attributes not expected to change between obs times). The set of metadata attributes of a time series (defined in TSMetadata in datastore.proto) doesn't include any position.

**Proposal:** Include an _optional_ position in TSMetadata (defined exactly as the one in ObsMedatata) that is set if and only if the time series is stationary. Strictly speaking, the observations of a stationary time series then wouldn't then need to have the position also in ObsMetadata, but it's probably convenient to keep them there as well (and the savings are likely to be insignificant (?)).

Note that the proposal assumes that the classification into either stationary or mobile can always be made at the ingestion level. If the ingestor (or ultimately the data producer) does not have that information, we must probably assume that the observation is mobile. Or ?

**Comment by lukas-phaf** [comment]
I agree that it would be helpful to be able to do a call that gives back all stationary time series. For example `/locations` endpoint in EDR needs this.

I see three options to implement this:
1. Leave the store as is, and the query decides on the fly which timeseries are stationary (by looking at the location of the corresponding observations).
2. The datastore handles this as you describe above, but the ingestion (user of PutObservations) doesn't have to deal with this. Instead, the datastore checks in PutObservation if it (still) holds that the timeseries is stationary, and set the optional position field in the TSMetadata appropriately.
3. Let the ingestion decide on this.

I think what you suggest is option 3?

If we go the route of 2 or 3, I see several options of storing this:
- Have an optional explicit position field in TSMetadata, and keep the mandatory one in ObsMetadata.
- Have an optional explicit position field in TSMetadata, and make the one in ObsMetadata optional. But what happens if we "change our mind" on this thing being stationary, in particular if we implement option 2 above).
- In TSMetadata just have a `is_stationary` boolean, and keep the mandatory one in ObsMetadata. If we need the actual position of the timeserie, any of the ones in ObsMetadata will do (as they should all be the same, and this should be validated in PutObservations!).

One question I have on this, what happens if a "stationary" time series is moved to a new position. How do we handle this?

**Comment by jo-asplin-met-no** [comment]
When I think of it now, perhaps it will just add unnecessary complexity to the datastore (and potentially also the ingestor) to try to maintain mobile/stationary classification of time series. So maybe the first option is really best, but maybe it could be better supported by adding a `temporal_mode` argument to GetObservations that would be _interval_ by default (and thus make use of `temporal_interval`), but could optionally be set to _latest_ which would return the single latest observation for the matching time series?

Then the EDR API wouldn't have to retrieve all available observations to make sure it got at least one for each time series, and we would save both time and space (i.e. it would be more unlikely to exceed the gRPC message size limit etc.).

The question is how this would work for mobile observations? For many time series (such as the ones without WIGOS IDs etc.) the EDR API couldn't really tell if a time series is stationary or not. (E.g. the observations could be generated from a vehicle that is parked for several hours and then starts to move...!) But maybe it isn't a big problem?

**Comment by lukas-phaf** [comment]
I think having the option of GetObservations would be a simple (?) way to the data that is needed for the `/locations` endpoint. Another option is to have a separate endpoint which returns no data at all, but it would still need to come up with a lat/lon. But maybe worth waiting till the review of what is needed for the `/locations` is done?

This indeed means the `/locations` endpoint sees all timeseries as stationary, as the distinction can not be made based on one time point. But I guess that is okay for now.
